bigquery/google/cloud/bigquery/table.py

```diff
@@ -1081,6 +1072,46 @@ def _build_schema_resource(fields):
     return infos
 
 
+def _item_to_row(iterator, resource):
+    """Convert a JSON row to the native object.
+
+    .. note::
+
+        This assumes that the ``schema`` attribute has been
+        added to the iterator after being created, which
+        should be done by the caller.
+
+    :type iterator: :class:`~google.cloud.iterator.Iterator`
+    :param iterator: The iterator that is currently in use.
+
+    :type resource: dict
+    :param resource: An item to be converted to a row.
+
+    :rtype: tuple
+    :returns: The next row in the page.
+    """
+    return _row_from_json(resource, iterator.schema)
+
+
+# pylint: disable=unused-argument
```

Comment:
What if we made `iterator` optional in [iterator.py](https://github.com/GoogleCloudPlatform/google-cloud-python/blob/3d28a5ceb1cba844e11710c98eac7933fbb4e977/core/google/cloud/iterator.py#L214)?

It seems like this might be a common case in the future.